borg/testsuite/archiver.py

```diff
@@ -1327,6 +1327,28 @@ def test_fuse(self):
     def test_debug_put_get_delete_obj(self):
         pass
 
+    def test_strip_components_doesnt_leak(self):
+        self.cmd('init', self.repository_location)
+        self.create_regular_file('dir/file', contents=b"test file contents 123")
+        self.create_regular_file('dir/file2', contents=b"test file contents 345")
+        self.create_regular_file('skipped', contents=b"test file contents 567")
+        self.create_regular_file('skipped2', contents=b"test file contentsasdasd")
+        self.create_regular_file('skipped4', contents=b"sdfdsgdgfhttztu")
```

Comment:
if the contents does not matter, just have to be different, can we just have b"contents 1", b"contents 2", ...?

filenames: skipped1, skipped2, skipped3?